### PR TITLE
Update rustc-ap-syntax to 541.0

### LINKED ISSUE #1058: build from master failed
cargo +nightly build --release

```
 Compiling rustc-ap-arena v491.0.0
error[E0599]: no method named `cap` found for type `alloc::raw_vec::RawVec<T>` in the current scope
   --> /Users/chenxiang/.cargo/registry/src/mirrors.ustc.edu.cn-12df342d903acd47/rustc-ap-arena-491.0.0/lib.rs:102:47
    |
102 |                 self.start().add(self.storage.cap())
    |                                               ^^^ private field, not a method

error[E0599]: no method named `cap` found for type `alloc::raw_vec::RawVec<T>` in the current scope
   --> /Users/chenxiang/.cargo/registry/src/mirrors.ustc.edu.cn-12df342d903acd47/rustc-ap-arena-491.0.0/lib.rs:273:55
    |
273 |                     new_capacity = last_chunk.storage.cap();
    |                                                       ^^^ private field, not a method

error[E0599]: no method named `cap` found for type `alloc::raw_vec::RawVec<u8>` in the current scope
   --> /Users/chenxiang/.cargo/registry/src/mirrors.ustc.edu.cn-12df342d903acd47/rustc-ap-arena-491.0.0/lib.rs:408:55
    |
408 |                     new_capacity = last_chunk.storage.cap();
    |                                                       ^^^ private field, not a method

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0599`.
error: Could not compile `rustc-ap-arena`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```
```
 (master)$ rustup --version
rustup 1.17.0 ( )
 (master)$ cargo --version
cargo 1.36.0 (c4fcfb725 2019-05-15)
 (master)$ rustc --version
rustc 1.36.0 (a53f9df32 2019-07-03)
 (master)$ rustup run nightly rustup show
Default host: x86_64-apple-darwin

installed toolchains
--------------------

stable-x86_64-apple-darwin
nightly-x86_64-apple-darwin

active toolchain
----------------

nightly-x86_64-apple-darwin (environment override by RUSTUP_TOOLCHAIN)
rustc 1.38.0-nightly (890881f8f 2019-07-25)
```

### PR DESCRIPTION
**Pull request by kngwyu**
Resolves #1058 
cc: @Xanewok @topecongiro

### CODE REVIEW

File: src/racer/ast.rs [hunk 1]
```diff
@@ -1326,12 +1327,14 @@ pub(crate) struct IfLetVisitor<'r, 's: 'r> {
 impl<'ast, 'r, 's> visit::Visitor<'ast> for IfLetVisitor<'r, 's> {
     fn visit_expr(&mut self, ex: &'ast ast::Expr) {
         match &ex.node {
-            ExprKind::IfLet(pats, expr, _, _) | ExprKind::WhileLet(pats, expr, _, _) => {
-                if let Some(pat) = pats.get(0) {
-                    self.let_pat = Some(Pat::from_ast(&pat.node, &self.scope));
-                    let mut expr_visitor = ExprTypeVisitor::new(self.scope.clone(), self.session);
-                    expr_visitor.visit_expr(expr);
-                    self.rh_expr = expr_visitor.result;
+            ExprKind::If(let_stmt, ..) => {
```
Comment (Xanewok) [inline src/racer/ast.rs]: Missing `ExprKind::While` or should this not process `while let`s?

Comment (kngwyu) [reply src/racer/ast.rs]: Good catch :+1:, thanks

**Comment by topecongiro** [comment]
@kngwyu My apologies for annoyance, but could you use 542.0.0, please? I have just released rustfmt 1.4.1 w/ 542.0.0 to fix the build failure in the latest nightly.

**Comment by kngwyu** [comment]
@topecongiro 
Thank you for releasing, I've just updated it.

**Comment by Xanewok** [comment]
Awesome, thank you :100: Could you cut a new release with the v542 bump? I'll use that to upgrade Racer and rustfmt to a single version in the RLS.

**Comment by kngwyu** [comment]
Sure, I just opened #1061

**Comment by kngwyu** [comment]
@Xanewok 
Just published 2.1.24

**Comment by Xanewok** [comment]
Awesome, thank you!

**Comment by jiacai2050** [comment]
@kngwyu Thanks. But there are still errors when build from master `7c313b3a142726b66006dae49b0c8d67b2877d70`

```
error[E0425]: cannot find function `type_name` in module `any`
   --> /Users/chenxiang/.cargo/registry/src/mirrors.ustc.edu.cn-12df342d903acd47/rustc-ap-serialize-542.0.0/serialize.rs:852:21
    |
852 |                any::type_name::<S>(),
    |                     ^^^^^^^^^ not found in `any`
help: possible candidates are found in other modules, you can import them into scope
    |
7   | use core::intrinsics::type_name;
    |
7   | use std::intrinsics::type_name;
    |

error[E0425]: cannot find function `type_name` in module `any`
   --> /Users/chenxiang/.cargo/registry/src/mirrors.ustc.edu.cn-12df342d903acd47/rustc-ap-serialize-542.0.0/serialize.rs:854:21
    |
854 |                any::type_name::<T>(),
    |                     ^^^^^^^^^ not found in `any`
help: possible candidates are found in other modules, you can import them into scope
    |
7   | use core::intrinsics::type_name;
    |
7   | use std::intrinsics::type_name;
    |

   Compiling syn v0.15.42
error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0425`.
```

**Comment by kngwyu** [comment]
@jiacai2050
You can reproduce it with the latest nightly?

**Comment by jiacai2050** [comment]
@kngwyu rustc 1.38.0-nightly (4560cb830 2019-07-28) compile fine. Many thanks.

**Pull request merged**